gnd/src/main/java/com/google/android/gnd/repository/DataRepository.java

```diff
@@ -158,82 +153,37 @@ private void onProjectLoaded(Project project) {
+   * <ol>
+   *   <li>Attempt to sync remote record changes to the local data store. If network is not
+   *       available or operation times out, this step is skipped.
+   *   <li>Relevant records are returned directly from the local data store.
+   * </ol>
    */
-  public Flowable<ImmutableList<Record>> getRecordSummariesOnceAndStream(
+  public Single<ImmutableList<Record>> getRecords(
       String projectId, String featureId, String formId) {
     // TODO: Only fetch first n fields.
-    // TODO: Also load from db.
     // TODO(#127): Decouple feature from record so that we don't need to fetch record here.
     return getFeature(projectId, featureId)
         .switchIfEmpty(Single.error(new DocumentNotFoundException()))
-        .flatMapPublisher(
-            feature -> {
-              Flowable<RemoteDataEvent<Record>> remoteChanges =
-                  remoteDataStore
-                      .loadRecordSummariesOnceAndStreamChanges(feature)
-                      .subscribeOn(Schedulers.io());
-              Flowable<ImmutableList<Record>> localChanges =
-                  localDataStore
-                      .getRecordsOnceAndStream(feature, formId)
-                      .subscribeOn(Schedulers.io());
-              return maybeSyncFirst(remoteChanges)
-                  .toFlowable()
-                  .map(o -> (ImmutableList<Record>) o)
-                  .concatWith(
-                      // This will rewrite the first update in remoteChanges to the local db
-                      // even if already successful. We allow this in case the first update
-                      // timed out or failed or the network wasn't available.
-                      localChanges.mergeWith(
-                          remoteChanges.flatMapCompletable(this::mergeRemoteRecordChange)));
-            });
+        .flatMap(feature -> getRecords(feature, formId));
   }
 
-  /**
-   * If network is available, waits for records from remote db, merging them into the local db
-   * before completing. Completes immediately if the network isn't available or if record sync fails
-   * or times out.
-   */
-  private Completable maybeSyncFirst(Flowable<RemoteDataEvent<Record>> remoteChanges) {
-    if (networkManager.isNetworkAvailable()) {
-      Log.d(TAG, "Has network; syncing records before showing");
-      return remoteChanges
-          .firstElement()
-          .flatMapCompletable(this::mergeRemoteRecordChange)
-          .timeout(GET_REMOTE_RECORDS_TIMEOUT_SECS, TimeUnit.SECONDS)
-          // TODO: Propagate this to the user so we can show "network unavailable" message also
-          // when network became unavailable mid-fetch .
-          .doOnError(t -> Log.d(TAG, "Record sync timed out"))
-          .onErrorComplete();
-    } else {
-      Log.d(TAG, "No network; skipping remote sync");
-      return Completable.complete();
-    }
+  private Single<ImmutableList<Record>> getRecords(Feature feature, String formId) {
+    Completable remoteSync =
```

Comment:
No more network check? Or did we move it elsewhere?

Seems like it's not strictly necessary from a functional perspective, but I wonder if we want to differentiate between timeouts and network absence in case we want to propagate that info upwards?

Reply:
Good catch. When we disable local caching via Firebase, it will fail fast and this request will fall back to the local db. We can add def propagation of this state once needed.

Reply:
sounds good!